README.md

```diff
@@ -1 +1,141 @@
+Do not reuse names. Do not use names that can mean multiple things. Always use the same name for the same thing. So use the same name within the JavaScript, CSS and HTML for the same thing. Align naming with the whole team: Backend, Frontend, UX, Design, PM, client, etc.
+
+For big projects with their own jargon it could help to create a dictionary.
+
+#### Descriptive
+Names must be descriptive for the working or usage of the class, method or variable.
+
+###### The name should always end with what the object is.
+For example: a button must always end with `Button`.
+ * `MenuButton` a button used in the menu
+ * `ButtonMenu` a menu with buttons
+ * `MenuButtonIcon` an icon in the button of the menu
+ 
+There is no limit for the length of a name, so prefer a long name which is clear and descriptive than a short name which is not clear.
+
+#### Abbreviations
+Avoid them as a general rule. For example, `calculateOptimalValue()` is a better method name than `calcOptVal()`. Being clear is more important than minimizing keystrokes. If you don't abbreviate, developers won't have to remember whether you shortened a word like _qualified_ to `qual` or `qlfd`.
+
+Abbreviations that are part of the HTML, CSS and/or JavaScript API are allowed, like:
+ * `URL` for Uniform Resource Locator
+ * `URI` for Uniform Resource Identifier
+ * `src` for source
+ * `DOM` for Document Object Model
+ * `img` for image
+ 
+You should only use these abbreviations within the same context. So only use `img` if you refer to an HTML Image Tag. (`<img>`)
+
+We have standardized on a few abbreviations that are allowed to use:
+ * `auto` for automatic, as in `autoLayout`
+ * `bin` for binary
+ * `info` for information, as in `GridRowInfo`
+ * `min` for minimum, as in `minWidth`
+ * `max` for maximum, as in `maxHeight`
+ * `regExp` for regular expression, as in `RegExpValidator`
+ * `util` or `utils` for utility or utilities, as in `StringUtils`
+ * `id` for identifier. Please note that 'd' should be written in lowercase when used in combination with an other word, like `userId`.
+ * `init` for initialize
+ * `ui` for user interface
+ * `temp` for temporary
+ * `app` for application
+ * `lib` for library
+ * `fps` for frames per second
+ * `args` for arguments
+ * `param` for parameter
+ * `params` for parameters
+ * `ref` for reference
+
+#### Plural or singular?
+
+##### Classes
+When you would instantiate or type the class it must have singular name, also for enums.
+When you should not instantiate the class, it should have a plural name.
+
+##### Arrays
+or other kind of lists should have a plural name or end with `List`, like `userList`.
+
+##### Folders
+If a folder holds multiple files, but all related to one main type, it should have a singular name. If it holds multiple main files of a type, it should have a plural name.
+
+For example, the folder `page` contains a single page, with maybe some helper files. The folder `pages` contains multiple pages.   
```

Comment:
config(uration) is already an umbrella/summation term on its own, it is conceptually not limited to a single entity, so it would (imo) only need to be plural if you have configuration-files for multiple 'configured things'.

Reply:
If a folder has the name `config` or `configurations` would mean that there is just one configuration, but (maybe) described in multiple files.

If a folder has the name `configurations` it means that there are multiple configurations in that folder and that it is (maybe) possible to switch.

So I think there is a big difference here in using a plural or singular name.